src/main/java/edu/iu/uits/lms/lti/service/OidcTokenUtils.java

```diff
@@ -71,71 +80,184 @@ public OidcTokenUtils(Map<String, Object> attributeMap) {
+      String roleStr = jsonObj.getAsString(CUSTOM_INSTRUCTURE_MEMBERSHIP_ROLES_KEY);
+      return null2Empty(StringUtils.split(roleStr, ","));
+   }
+
+   /**
+    * Get filtered Instructure Membership Roles from the Custom claim
+    * @return Filtered Instructure Membership Roles
+    */
+   public String[] getCustomInstructureMembershipRoles() {
+      String[] rawRoles = getCustomInstructureMembershipRolesRaw();
+      return cleanRoleList(rawRoles);
+   }
+
+   /**
+    * Get the GUID out of the Platform claim
+    * @return Platform GUID
+    */
    public String getPlatformGuid() {
       JSONObject jsonObj = (JSONObject) attrMap.get(Claims.PLATFORM_INSTANCE);
       return jsonObj.getAsString(CLAIMS_PLATFORM_GUID_KEY);
    }
 
+   /**
+    * Get the specified key out of the Custom claim
+    * @param key Key
+    * @return Value of given key from the Custom claim
+    */
    public String getCustomValue(String key) {
       JSONObject jsonObj = (JSONObject) attrMap.get(Claims.CUSTOM);
       return jsonObj.getAsString(key);
    }
 
+   /**
+    * Get the specified key out of the Context claim
+    * @param key Key
+    * @return Value of given key from the Context claim
+    */
    public String getContextValue(String key) {
       JSONObject jsonObj = (JSONObject) attrMap.get(Claims.CONTEXT);
       return jsonObj.getAsString(key);
    }
 
+   /**
+    * Get the specified key out of the LIS claim
+    * @param key Key
+    * @return Value of given key from the LIS claim
+    */
    public String getLisValue(String key) {
       JSONObject jsonObj = (JSONObject) attrMap.get(Claims.LIS);
       return jsonObj.getAsString(key);
    }
 
+   /**
+    * According to the LTI spec, when a user has the TA role, since it is a sub-type of Instructor, both roles are sent.
+    * This makes it impossible to distinguish between the two.  So, this will strip out the "extra" occurrence of the Instructor role.
+    * It should still support legit cases where a user is a TA in one section and an Instructor in a different section.
+    * @param listArray Input array of roles
+    * @return Filtered array or roles
+    */
+   protected String[] cleanRoleList(String[] listArray) {
+      List<String> list = new ArrayList<>(Arrays.asList(listArray));
+      log.debug("Original list: {}", list);
+
+      // Group by the counts for each role
+      Map<Object, Long> collect = list.stream().collect(Collectors.groupingBy(l -> l, Collectors.counting()));
+      log.debug("Grouped: {}", collect);
+
+      // If there is at least 1 TA role and 1 Instructor role, remove the instructor role
+      if (collect.containsKey(LTIConstants.CANVAS_TA_ROLE) && collect.get(LTIConstants.CANVAS_TA_ROLE) > 0 &&
+            collect.containsKey(LTIConstants.CANVAS_INSTRUCTOR_ROLE) && collect.get(LTIConstants.CANVAS_INSTRUCTOR_ROLE) > 0) {
+         log.info("Remove a {}...", LTIConstants.CANVAS_INSTRUCTOR_ROLE);
+         list.remove(LTIConstants.CANVAS_INSTRUCTOR_ROLE);
```

Comment:
Might need multiple removes if there happen to be multiple TAs in the list

Reply:
Best indications are that "duplicates" aren't sent in the data.  Only one role is given for any number of actual instances